OOP/LibraryManagement/LibraryApp/Services/DocumentCache.cs

```diff
@@ -0,0 +1,43 @@
+﻿using LibraryApp.Documents;
+using Microsoft.Extensions.Caching.Memory;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace LibraryApp.Services;
+
+public class DocumentCache(Dictionary<DocumentType, TimeSpan> cacheExpiryPerType)
+{
+    private readonly MemoryCache _cache = new(new MemoryCacheOptions());
+    private readonly Dictionary<DocumentType, TimeSpan> _cacheExpiryPerType = cacheExpiryPerType;
+
+    public T GetDocument<T>(string title) where T : Document
+    {
+        _cache.TryGetValue(title, out T document);
+        
+        return document;
+    }
+
+    public void SetDocument<T>(T document) where T : Document
+    {
+        if (_cacheExpiryPerType.TryGetValue(document.Type, out var expiry))
+        {
+            if (DateTime.Now + expiry > DateTime.Now)
```

Comment:
1. Let's reduce, please, extra nesting here as well
2. All server apps should not work with local date time (there are some really rare specific conditions to do that). Instead, use UTC to make sure that whenever region your app is hosted it will always use the same comparison time